### ISSUE TITLE
Bad running in Jupyter Notebook

### ISSUE THREAD

**Issue by AlirezaBoroon** [microsoft/vscode#137448]
Issue Type: <b>Bug</b>

Hi
I have a problem with running program in Jupyter NoteBook, when running, sometimes the VS Code doesn't show the printed stuff at the current moment, and if I go forward and one step forward, it will show the previous printed line, and I can't see that line, at that correct moment, and I have to go and one step forward in the program running to see the details of the previous step.
Therefore please solve this issue.
The issue video URL:
https://mega.nz/file/2uB0SRQA#QiRSf96dmpa4A0Dx1YfqlPBAJig5SgSJlTxzRVcnYSE
(Once I typed this issue and I didn't get any answer)
Thanks.

VS Code version: Code 1.62.2 (3a6960b964327f0e3882ce18fcebd07ed191b316, 2021-11-11T20:56:38.428Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz (8 x 1800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|6.18GB (1.29GB free)|
|Process Argv|--crash-reporter-id 27f72fee-8da4-4129-bb28-02f0269ce2ab|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
theme-monokai-pro-vscode|mon|1.1.19
python|ms-|2021.11.1422169775
vscode-pylance|ms-|2021.11.2
jupyter|ms-|2021.10.1101450599
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.3
polacode|pnp|0.3.4
material-theme|zhu|3.13.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30384385
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
dockerwalkthru:30377721
bridge0723:30353136
pythonrunftest32:30373476
pythonf5test824:30373475
javagetstartedt:30391933
pythonvspyt187:30373474
vsqsis200:30402083
vsaa593:30376534
vssld246cf:30401891

```

</details>

**Comment by roblourens** [comment]
Sorry, I really don't understand the problem. Can you explain in more detail?

**Comment by AlirezaBoroon** [comment]
That's all, why you guys keep saying I don't understand?, I even uploaded a video for you.
Is the mistake from my English grammar? or you don't try to understand?

**Comment by roblourens** [comment]
I don't understand what the video is showing, and I don't understand what you mean by "one step forward/back". Please explain what you are expecting to happen

**Comment by AlirezaBoroon** [comment]
Now good to hear it, you want me to say exactly what you didn't understand, 
When my code is running, it wants an input for continuing, so when I enter a manual input some texts should be printed after that for example say thanks for input, but we can't see it printed, OK? but actually they have been executed, after that lines, in my code, it gets an input again and then another printed stuff, so if I enter the next input (this means one step forward) then in the output section, NOW we see the previous lines that were executed but not printed and shown, with the lines printed at the current moment, that's the problem, and what we are expecting to be is to see the printed lines at their correct moment (the time that they should be in the output).
Thanks
And let me know where you didn't get.

**Comment by roblourens** [comment]
Can you share the notebook?

**Comment by AlirezaBoroon** [comment]
`def includedornot(l, s):
    for i in range(0, len(s)):
        if s[i]== l:
            return 1
    return 0
def removeLetter(l, s):
    sl= []
    for i in range(0, len(s)):
        sl.append(s[i])
    for j in range(0, len(sl)):
        if sl[j]== l:
            sl[j]=""
    return sl
def main():
    print("Please enter the stirng you want to remove a letter from:")
    string= input()
    print("Now please enter the letter you want to be removed from the string you entered:")
    while True:
        letter= input()
        firstconfirm= includedornot(letter, string)
        if firstconfirm== 1:
            break
        elif firstconfirm== 0:
            print("Please enter again, the letter is not in the string!:")
            continue
    finalstring= removeLetter(letter, string)
    print("The final string without the letter", letter, "is:")
    fins= ""
    for i in range(0, len(finalstring)):
        fins+= finalstring[i]
    print(fins)
    print("End of the program.")
    return
#start0: # welcome and goodbye
print("Hello and welcome, this program is used for you to be able to delete a particular letter from a string you have entered.")
main()
while True:
    print("Do you want to start again? 1 for yes / 0 for no")
    choice= int(input())
    if choice== 1:
        main()
    elif choice== 0:
        print("Thanks for using the app.")
        break
print("Goodbye.")`
------------------------------------------------------------------
This is the code which I showed in the video as well.

**Comment by roblourens** [comment]
I get it now, thanks. 
Can you try downloading the Insiders' build and tell me whether it happens there? https://code.visualstudio.com/insiders

**Comment by AlirezaBoroon** [comment]
@roblourens Yes when you told that, I installed the Insiders app and I tried the Insiders build, but nothing happened and the issue isn't solved, I also tried update the python to version 3.10.0, but the issue's still happening.

**Comment by roblourens** [comment]
Any idea how to troubleshoot it @rebornix, an output from `print("...")` is not showing up at the right time until some more output is added

**Comment by roblourens** [comment]
ping @rebornix

**Comment by AlirezaBoroon** [comment]
Hi, the issue is still happening and the team has not yet solved it, I think you are responsible for this, if you guys don't know what to do, please refer to someone else.

**Comment by roblourens** [comment]
Sorry for the slow response but lots of changes in the past year. Please let me know if it's still an issue
